MediaBrowser.Model/Dlna/StreamBuilder.cs

```diff
@@ -936,22 +937,32 @@ private void BuildStreamVideoItem(
             }
 
             // Prefer matching audio codecs, could do better here
-            var audioCodecs = ContainerProfile.SplitValue(audioCodec);
+            List<string> audioCodecs = [];
+            if (audioCodec is not null)
+            {
+                audioCodecs.AddRange(ContainerHelper.Split(audioCodec));
+            }
+
+            if (audioCodecs.Count == 0 && audioStream is not null)
+            {
+                // Add the original codec if no codec is specified
+                audioCodecs.Add(audioStream.Codec);
+            }
 
             // Enforce HLS audio codec restrictions
             if (playlistItem.SubProtocol == MediaStreamProtocol.hls)
             {
                 if (string.Equals(playlistItem.Container, "mp4", StringComparison.OrdinalIgnoreCase))
                 {
-                    audioCodecs = audioCodecs.Where(codec => _supportedHlsAudioCodecsMp4.Contains(codec)).ToArray();
+                    audioCodecs = audioCodecs.Where(codec => _supportedHlsAudioCodecsMp4.Contains(codec)).ToList();
                 }
                 else
                 {
-                    audioCodecs = audioCodecs.Where(codec => _supportedHlsAudioCodecsTs.Contains(codec)).ToArray();
+                    audioCodecs = audioCodecs.Where(codec => _supportedHlsAudioCodecsTs.Contains(codec)).ToList();
                 }
             }
 
-            var audioStreamWithSupportedCodec = candidateAudioStreams.Where(stream => ContainerProfile.ContainsContainer(audioCodecs, stream.Codec)).FirstOrDefault();
+            var audioStreamWithSupportedCodec = candidateAudioStreams.Where(stream => playlistItem.AudioCodecs.Contains(stream.Codec, StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
```

Comment:
To reuse the codec comparison logic. `playlistItem.AudioCodecs` is also not yet populated.
```suggestion
            var audioStreamWithSupportedCodec = candidateAudioStreams.Where(stream => ContainerHelper.ContainsContainer(audioCodecs, stream.Codec)).FirstOrDefault();
```